### PR TITLE
Adding SDK code for AzureAnalysisServices

### PR DESCRIPTION
**Pull request by athipp**
This project is for enabling client code generation for Azure Analysis
services. The swagger spec for that is here
https://github.com/Azure/azure-rest-api-specs/pull/579#issuecomment-251185810

https://github.com/Azure/azure-rest-api-specs/blob/master/arm-asazure/2016-05-06/swagger/asazure.json

This checkin contains
- Generatedcode for the swagger spec
- InmemoryTests for the APIs.
  Note: For now we have sufficient coverage with inmemroy tests. I would
  like to go ahead with powershell development. So scenario tests is
  coming in further checkin.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+        {
+            var acceptedResponse = new HttpResponseMessage(HttpStatusCode.Created)
+            {
+                Content = new StringContent(AnalysisServicesTestUtilities.GetDefaultCreatedResponse("InProgress"))
+            };
+
+            acceptedResponse.Headers.Add("x-ms-request-id", "1");
+            acceptedResponse.Headers.Add("Location", @"http://someLocationURL");
+
+            var okResponse = new HttpResponseMessage(HttpStatusCode.OK)
```
Comment (shahabhijeet) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs]: >HttpResponseMessage [](start = 33, length = 19)

Any reason you are not using HttpRecorder to record your tests?

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            return client;
+        }
+
+        public static AnalysisServicesClient GetAnalysisServicesClient(MockContext context, RecordedDelegatingHandler handler)
+        {
+            AnalysisServicesClient analysisServicesClient;
+            if (IsTestTenant)
+            {
+                analysisServicesClient = new AnalysisServicesClient(new TokenCredentials("xyz"), GetHandler());
+                analysisServicesClient.SubscriptionId = testSubscription;
```
Comment (shahabhijeet) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs]: >testSubscription [](start = 56, length = 16)

Any reason you are not using TestEnvironment and connection string to set all your credentials and subscription information?

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs (hunk 1 above)
Comment (athipp) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs]: I am working on it and will submit a separate PR for that. SPN based authentication doesn;t work so I need to figure out how to do username based auth. I want to make progress on powershell code though. thats why I submitted this.

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs (hunk 2 above)
Comment (athipp) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs]: I am working on it and will submit a separate PR for that. SPN based authentication doesn;t work so I need to figure out how to do username based auth. I want to make progress on powershell code though. thats why I submitted this first

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+            return rgname;
+        }
+
+        public static void DeleteResourceGroup(ResourceManagementClient resourcesClient, string resourceGroupName)
+        {
+            //if (!IsTestTenant)
+            //{
+            //    resourcesClient.ResourceGroups.Delete(resourceGroupName);
+            //}
```
Comment (shahabhijeet) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs]: Remove commented code

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// Version information for an assembly consists of the following four values:
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
```
Comment (shahabhijeet) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Properties/AssemblyInfo.cs]: remove commented code

**Comment by shahabhijeet** [comment]
> using Microsoft.Azure.Management.DataLake.Store;

I think this might be a mistake, can you revert this file and not combine with your PR

---
Refers to: src/ResourceManagement/DataLake.Store/DataLakeStore.Tests/ScenarioTests/AccountOperationTests.cs:16 in 8b97eea. [](commit_id = 8b97eeaa35da64c32ab317a8b8bb0a2ad7667d23, deletion_comment = False)

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/ScenarioTests/ServerOperationsTests.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+using System.Linq;
+using System.Net;
+using Xunit;
+
+namespace AnalysisServices.Tests.ScenarioTests
+{
+    public class ServerOperationsTests : TestBase
+    {
+        [Fact]
+        public void CreateGetUpdateDeleteTest()
```
Comment (shahabhijeet) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/ScenarioTests/ServerOperationsTests.cs]: >CreateGetUpdateDeleteTest [](start = 20, length = 25)

Can this be split into multiple smaller tests? If possible.
For e.g. can the update/delete be separate tests reusing the create from a common code.
this is just a guidance, because in case if the test failed, you won't know if the create failed or update or deleting the account failed.

**Comment by athipp** [comment]
Closing this PR for now. Will open it once comments are fixed.

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (athipp) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Properties/AssemblyInfo.cs]: done

File: src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/ScenarioTests/ServerOperationsTests.cs (hunk 5 above)
Comment (athipp) [inline src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/ScenarioTests/ServerOperationsTests.cs]: For this release I would like to go with one test that test the happy path.

**Comment by azurecla** [comment]
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;
